graphstorm-processing/graphstorm_processing/config/config_parser.py

```diff
@@ -0,0 +1,383 @@
+
+from graphstorm_processing.constants import SUPPORTED_FILE_TYPES
+from .label_config_base import LabelConfig, EdgeLabelConfig, NodeLabelConfig
+from .feature_config_base import FeatureConfig, NoopFeatureConfig
+from .data_config_base import DataConfig
+
+
+def parse_feat_config(feature_dict: Dict) -> FeatureConfig:
+    """Parses a feature configuration, converting it to the correct type of feature config.
+
+    Parameters
+    ----------
+    feature_dict : Dict
+        A feature configuration dictionary in GSProcessing format
+
+    Returns
+    -------
+    FeatureConfig
+        A feature configuration object
+
+    Raises
+    ------
+    RuntimeError
+        _description_
+    """
+    # Special case: missing transformation is parsed as a no-op
+    if "transformation" not in feature_dict:
+        feature_dict["transformation"] = {"name": "no-op"}
+        return NoopFeatureConfig(feature_dict)
+
+    transformation_name = feature_dict["transformation"]["name"]
+
+    if transformation_name == "no-op":
+        return NoopFeatureConfig(feature_dict)
+    else:
+        raise RuntimeError(f"Unknown transformation name: '{transformation_name}'")
+
+
+class StructureConfig(ABC):
+    """
+    Base class for edge and node configurations
+    """
+
+    def __init__(self, data_dict: Dict[str, Any]):
+        self._data_config = DataConfig(
+            format=data_dict["format"],
+            files=data_dict["files"],
+            separator=data_dict.get("separator", None),
+        )
+        self._files = self._data_config.files
+        self._separator = self._data_config.separator
+        self._format = self._data_config.format
+        self._feature_configs: List[FeatureConfig] = []
+        self._labels: List[LabelConfig] = []
+
+    @property
+    def data_config(self) -> DataConfig:
+        """
+        The data configuration for the structure.
+        """
+        return self._data_config
+
+    @property
+    def files(self) -> Sequence[str]:
+        """
+        The list of the files to be parsed.
+        """
+        return self._files
+
+    @property
+    def separator(self) -> Optional[str]:
+        """
+        The separator of the files. None if the input files are in Parquet format.
+        """
+        return self._separator
+
+    @property
+    def format(self) -> str:
+        """
+        The format of the files.
+        """
+        return self._format
+
+    @property
+    def label_configs(self) -> Optional[List[LabelConfig]]:
+        """
+        The list of labels, if they exist, None otherwise.
+        """
+        return self._labels if self._labels else None
+
+    @property
+    def feature_configs(self) -> Optional[List[FeatureConfig]]:
+        """
+        The list of labels, if they exist, None otherwise.
+        """
+        return self._feature_configs if self._feature_configs else None
+
+    def merge_same_type_features(self) -> None:
```

Comment:
What is the purpose of this function?
Why it is empty?
Shall we raise NoImplemented error?